Jamulus.pro

```diff
@@ -203,6 +203,10 @@ win32 {
     LIBS += -framework AVFoundation \
         -framework AudioToolbox
 } else:android {
+    ANDROID_ABIS = armeabi-v7a arm64-v8a x86 x86_64
+    ANDROID_VERSION_NAME = $$VERSION
+    ANDROID_VERSION_CODE = $$system(git log --oneline | wc -l)
+
```

Comment:
I couldn't see in the build log where these get used. If they are used silently, then perhaps they could be echoed too?

Reply:
Yep, they're qmake-internal variables. I've added a `message()` call to output them.

Reply:
Just tried installing jamulus_3.8.2dev-5f108fd_android.apk over 3.8.0 ... result is as before, i.e. version number is reported as being different (see screenshot attached), but install ends with message "App not installed". I suspect this is "as expected" at the moment! BTW, if you want me to check any functionality on any new build let me know - I have several Android devices at my disposal, and can force a new build to install by uninstalling any old one, of course.
![Screenshot_20220222-091013_Total Commander](https://user-images.githubusercontent.com/85576919/155102345-b30c093c-f1ea-40a2-a414-8a98bd2015cb.jpg)

Reply:
Sorry, first time I've posted an image - didn't realise it'd come out that big on screen!

Reply:
(I've edited your comment to use `<img src="...existing url.." width="2002>`to reduce the size a bit.

> result is as before, i.e. version number is reported as being different (see screenshot attached)
> but install ends with message "App not installed". I suspect this is "as expected" at the moment!

Thanks for the screenshot. Yes, seeing the real version number here is definitely an improvement. And yes, it's expected that this PR does not solve all update problems on Android. While I had hoped for this to be the case, the PR doesn't claim to do it :)

> BTW, if you want me to check any functionality on any new build let me know - I have several Android devices at my disposal, and can force a new build to install by uninstalling any old one, of course.

Great, thanks! I'll likely ask again as part of #2404.